src/diffusers/pipelines/flux/pipeline_flux.py

```diff
@@ -417,6 +418,23 @@ def check_inputs(
         if max_sequence_length is not None and max_sequence_length > 512:
             raise ValueError(f"`max_sequence_length` cannot be greater than 512 but is {max_sequence_length}")
 
+        if latents is not None:
+            if not isinstance(latents, torch.Tensor):
+                raise ValueError(f"`latents` has to be of type `torch.Tensor` but is {type(latents)}")
+ 
+            batch_size, num_patches, channels = latents.shape
```

Comment:
Wouldn't this throw an error if the shape is `ndim==4`. Perhaps we check to see if `ndim==3` here and raise and error rather than letting it fail.